components/ratgdo/ratgdo.cpp

```diff
@@ -222,10 +235,53 @@ namespace ratgdo {
                 this->send_command(Command::GET_STATUS);
             }
             ESP_LOGD(TAG, "Motion: %s", MotionState_to_string(*this->motion_state));
-        } else if (cmd == Command::SET_TTC) {
+        } else if (cmd == Command::TTC_SET_DURATION) {
             auto seconds = (byte1 << 8) | byte2;
-            ESP_LOGD(TAG, "Time to close (TTC): %ds", seconds);
-        }
+            ESP_LOGD(TAG, "Time to close (TTC) update request: %ds", seconds);
+        } else if (cmd == Command::TTC_DURATION) {
+            auto seconds = (byte1 << 8) | byte2;
+            ESP_LOGD(TAG, "Time to close (TTC) set to: %ds", seconds);
+            query_status_flags_ |= QSF_TCC_DUR;
+            if (seconds == 60 || seconds == 300 || seconds == 600 || seconds == 0) {
+                this->ttc_time_seconds=seconds;
+            } else if (seconds != 1) {
+                this->ttc_time_seconds=0;
+                ESP_LOGW(TAG, "Unsupported TTC time: %ds", seconds);
+            }
+            if(this->restore_hold_state_  == true && this->restore_ttc_ == false) {
+                this->hold_enable();
+                this->restore_hold_state_  = false;
+            } 
+        } else if (cmd == Command::TTC_COUNTDOWN) {
+            auto seconds = (byte1 << 8) | byte2;
+            ESP_LOGD(TAG, "(TTC) door will close in: %ds", seconds);
+        } else if (cmd == Command::TTC_CANCEL) {
+            if (byte1 == 0x04) {
+                ESP_LOGD(TAG, "TTC: Auto Hold Toggled");
+            } else if (byte1 == 0x05) {
+                ESP_LOGD(TAG, "TTC: Disabled");
+            } else {
+                ESP_LOGD(TAG, "TTC_CANCEL: Unknown Data");
+            }
+        } else if (cmd == Command::EXT_STATUS) {
+            query_status_flags_ |= QSF_EXT_STATUS;
+            if ( (byte1 & 0b00000111) == 0b00000001) {
+                ESP_LOGD(TAG, "TTC is disabled.");
+                this->hold_state = HoldState::HOLD_DISABLED;
+            } else if ( (byte1 & 0b00000111) == 0b00000010) {
+                ESP_LOGD(TAG, "TTC is enabled but in Hold Open.");
+                this->hold_state = HoldState::HOLD_ENABLED;
+            } else if ( (byte1 & 0b00000111) == 0b00000100) {
+                ESP_LOGD(TAG, "TTC is enabled.");
+                this->hold_state = HoldState::HOLD_DISABLED;
+            } else if ( (byte1 & 0b00000111) == 0b00000101) {
+                ESP_LOGW(TAG, "TTC closing was interrupted!  101");
+                this->hold_state = HoldState::HOLD_DISABLED;
+            } else if ( (byte1 & 0b00000111) == 0b00000110) {
+                ESP_LOGW(TAG, "TTC closing was interrupted!  110");
+                this->hold_state = HoldState::HOLD_DISABLED;
+            }
+        }            
```

Comment:
These need investigated a bit more.  Seems to be difference with door open vs. door closed.